packages/components/src/confirm/hook.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Internal dependencies
+ */
+import { useContextSystem, PolymorphicComponentProps } from '../ui/context';
+
+/**
+ * Internal dependencies
+ */
+import * as styles from './styles';
+import { useCx } from '../utils/hooks/use-cx';
+import type { OwnProps } from './types';
+
+export function useConfirm(
+	props: PolymorphicComponentProps< OwnProps, 'div' >
+) {
+	const { className, ...otherProps } = useContextSystem( props, 'Confirm' );
+
+	const cx = useCx();
+
+	const classes = cx( className );
+	const wrapperClassName = cx( styles.wrapper );
+	const overlayClassName = cx( styles.overlay );
+
+	return {
+		className: classes,
```

Comment:
`className` is not used by the component that calls this hook.

Also, this hook could be inlined directly into the component, with direct calls to `useContextSystem` and `useCx`.

Reply:
I copied the structure for an existing component for this draft, so there might be some code that's actually redundant. Thanks for pointing that out, though! Inlining makes sense 👍🏻